annotations/views/project_views.py

```diff
@@ -60,6 +60,30 @@ def add_text(self, request, pk=None):
         serializer = ProjectSerializer(project)
         return Response(serializer.data)
 
+    @action(detail=True, methods=['POST'], url_name='removetext')
+    def remove_text(self, request, pk=None):
+        text_id = request.data['text_id']
+        repo_id = request.data['repository_id']
+
+        repository = get_object_or_404(Repository, pk=repo_id)
+        project = get_object_or_404(TextCollection, pk=pk)
+        manager = repository.manager(request.user)
+        resource = manager.resource(id=int(text_id))
+
+        defaults = {
+            'title': resource.get('title'),
+            'created': resource.get('created'),
+            'repository': repository,
+            'repository_source_id': text_id,
+            'addedBy': request.user,
+        }
+        text, _ = Text.objects.get_or_create(uri=resource.get('uri'),
+                                             defaults=defaults)
+        project.texts.remove(text)
```

Comment:
Where is remove coming from? Did you define it somewhere?

Reply:
.remove is a django function. I found it in the docs https://docs.djangoproject.com/en/3.0/ref/models/relations/

Reply:
I did some reading on remove because I was curious. Remove only removes the relationship. So if object bar is related to object foo and you use remove on bar then both foo and bar still exist but bar is no longer related to object foo. While this creates part of the functionality that you intended it does not actually do want you want which is removing it from the database.